src/Options/Generic.hs

```diff
@@ -1,7 +1,11 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DataKinds                  #-}
 {-# LANGUAGE DefaultSignatures          #-}
 {-# LANGUAGE DeriveGeneric              #-}
 {-# LANGUAGE FlexibleInstances          #-}
 {-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE PolyKinds                  #-}
```

Comment:
Is there a way to do this without `PolyKinds`?  The reason I ask is that it deteriorates the quality of the instance lists for the haddocks.  If you do really need to do this I would like to split off the types that require this into a separate module so that types that don't need this are unaffected